recipes/azure-storage-cpp/all/conanfile.py

```diff
@@ -0,0 +1,121 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+class AzureStorageCppConan(ConanFile):
+    name = "azure-storage-cpp"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Azure/azure-storage-cpp"
+    description = "Microsoft Azure Storage Client Library for C++"
+    topics = ("azure", "cpp", "cross-platform", "microsoft", "cloud")
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compiler_version(self):
+        return {
+            "gcc": "5",
+            "Visual Studio": "14",
+            "clang": "3.4",
+            "apple-clang": "5.1",
+        }
+
+    def requirements(self):
+        self.requires("cpprestsdk/2.10.18")
+        if self.settings.os != "Windows":
+            self.requires("libxml2/2.9.10")
+            self.requires("libuuid/1.0.3")
+        if self.settings.os == "Macos":
+            self.requires("libgettext/0.20.1")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["CMAKE_FIND_FRAMEWORK"] = "LAST"
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.definitions["BUILD_SAMPLES"] = False
+        if not self.settings.compiler.cppstd:
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = self._minimum_cpp_standard
+
+        if self.settings.os == "Macos":
+            self._cmake.definitions["GETTEXT_LIB_DIR"] = self.deps_cpp_info["libgettext"].lib_paths[0]
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
```

Comment:
```suggestion
            del self.options.fPIC

   def validate(self):
```

Check minimal cppstd using validate method: https://docs.conan.io/en/latest/reference/conanfile/methods.html#validate